add instructions for use on Windows

robkam opened the pull request.

robkam committed:
add instructions for use on Windows

robkam committed:
Update README.md

robkam committed:
Update README.md

elsiehupp committed:
Add other OS-specific instructions (and generally tidy)

Signed-off-by: Elsie Hupp <github@elsiehupp.com>

elsiehupp committed:
Merge branch 'python3' into patch-1

elsiehupp commented:
Hi @robkam—thank you for thinking of this!

I took your idea and elaborated on it, and my changes are over in [a pull request on your fork](https://github.com/robkam/wikiteam3/pull/1). Once you merge the changes over there we can continue over here.

Because merging between forks creates complicated things like the second pull request, I've added you as a collaborator on this repository here, so you can create and edit forks on it, too. Given the fact that the branch you're merging here has the generic name `patch-1`, my guess is that you created it via the GitHub web interface, but once we've finished this pull request, I'll add comments explaining how to change your locally cloned copy of `wikiteam3` so that you can more easily create branches from there, too, which is useful if you're making changes to the Python code and not just the README.

robkam committed:
Merge pull request #1 from elsiehupp/pr/20

Add other OS-specific instructions (and generally tidy)

robkam committed:
Update README.md

On Windows this is the method that works for me, It ought to be verified. Possibly it's an unconventional approach. Having to open Bash as administrator is somewhat strange.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

robkam commented:
Hi @elsiehupp 

I understand how Git and GitHub work but still find them a little counterintuitive. Your instructions are very helpful.

At first I installed Python 3.10 via Microsoft Apps but then I uninstalled it and reinstalled it from Python.org. It's possible that some cruft left over from that has caused now having to open Bash as administrator.

elsiehupp committed:
Further expanding Windows instructions

Signed-off-by: Elsie Hupp <github@elsiehupp.com>

elsiehupp commented:
(See https://github.com/robkam/wikiteam3/pull/2)

elsiehupp commented:
> I understand how Git and GitHub work but still find them a little counterintuitive. Your instructions are very helpful.

Glad to hear! Part of what makes it counterintuitive is that Git is designed to be fully peer-to-peer and decentralized, and having a single, central host for a repository, i.e. GitHub, is sort of just a convenience built on top of that.

> At first I installed Python 3.10 via Microsoft Apps but then I uninstalled it and reinstalled it from Python.org. It's possible that some cruft left over from that has caused now having to open Bash as administrator.

My best guess is that the reason you're needing to run as Bash as an administrator is in order to be able to install scripts to Program Files. I don't think you would need to run as administrator if you were installing scripts to AppData instead.

elsiehupp commented:
Looking at the reviews for [Python on the Microsoft Store](https://apps.microsoft.com/store/detail/python-39/9P7QFQMJRFP7) it looks like Python not linking to `%PATH%` is specific to that version (as the python.org installer does so by default).

robkam reviewed (commented):
updating readme

robkam committed:
Merge pull request #2 from elsiehupp/pr/20-1

Further expanding Windows instructions

robkam committed:
Merge branch 'elsiehupp:python3' into patch-1
